src/corefx/System.Globalization.Native/localeStringData.cpp

```diff
@@ -240,3 +255,83 @@ extern "C" int32_t GetLocaleInfoString(const UChar* localeName, LocaleStringData
 
 	return UErrorCodeToBool(status);
 }
+
+/*
+Function:
+NormalizeTimePattern
+
+Convert an ICU non-localized time pattern to .NET format
+*/
+void NormalizeTimePattern(const UnicodeString *srcPattern, UnicodeString *destPattern)
+{
+	// An srcPattern example: "h:mm:ss a"
+	// A destPattern example: "h:mm:ss tt"
+	destPattern->remove();
+
+	for (int i = 0; i <= srcPattern->length() - 1; i++)
+	{
+		UChar ch = srcPattern->charAt(i);
+		switch (ch)
+		{
+		case ':':
+		case '.':
+		case 'H':
+		case 'h':
+		case 'm':
+		case 's':
+			destPattern->append(ch);
+			break;
+
+		case chSpace:
+		case chSpace2:
+			destPattern->append(chSpace);
+			break;
+
+		case 'a':
+			destPattern->append("tt");
+			break;
+		}
+	}
+}
+
+/*
+PAL Function:
+GetLocaleTimeFormat
+
+Obtains time format information.
+Returns 1 for success, 0 otherwise
+*/
+extern "C" int32_t GetLocaleTimeFormat(const UChar* localeName, int shortFormat, UChar* value, int32_t valueLength)
+{
+	Locale locale = GetLocale(localeName);
+	if (locale.isBogus())
+	{
+		return UErrorCodeToBool(U_ILLEGAL_ARGUMENT_ERROR);
+	}
+
+	DateFormat::EStyle style = (shortFormat != 0) ? DateFormat::kShort : DateFormat::kMedium;
+	LocalPointer<DateFormat> dateFormat(DateFormat::createTimeInstance(style, locale));
+	if (dateFormat == NULL || !dateFormat.isValid())
+	{
+		return UErrorCodeToBool(U_MEMORY_ALLOCATION_ERROR);
+	}
+
+	// cast to SimpleDateFormat so we can call toPattern()
+	SimpleDateFormat* sdf = dynamic_cast<SimpleDateFormat*>(dateFormat.getAlias());
```

Comment:
Eric and I had a conversation on his topics. Summary: cannot call SimpleDataFormat ctor directly so we need the cast etc, keep using DateFormat::short\medium instead of getBestPattern, add support to return the full and long times to managed code to put them into the full time array